From 3e4d40d18f08248d8d4fb0e1b7d650c673d7d28a Mon Sep 17 00:00:00 2001
From: KhabarlakKonstantin <habarlack@gmail.com>
Date: Tue, 3 Oct 2017 11:22:40 +0300
Subject: [PATCH 60/67] Fix crash after deleting a video on Albums tab

getMediaItem can return empty list if no more media items available,
break folder processing then
Number of items returned by getMediaItem can be inconsistent with
getMediaItemCount

Change-Id: I7f2abce5d3832a1b2c1fe068b629a3474ed742e2
---
 src/com/android/gallery3d/app/AlbumPage.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/com/android/gallery3d/app/AlbumPage.java b/src/com/android/gallery3d/app/AlbumPage.java
index 874670e83..4f3f34814 100644
--- a/src/com/android/gallery3d/app/AlbumPage.java
+++ b/src/com/android/gallery3d/app/AlbumPage.java
@@ -662,6 +662,9 @@ public class AlbumPage extends ActivityState implements GalleryActionBar.Cluster
             if (list == null) {
                 continue;
             }
+            if (list.isEmpty()) {
+                break;
+            }
             item = list.get(0);
             if (item == null) {
                 continue;
-- 
2.17.1

